postmortems/2021-07-27_update_ris3.md

```diff
@@ -0,0 +1,67 @@
+- Running sheet-to-triples with just the `--add-graph ris3_points.ttl` argument and no transforms means it doesn't write anything to file -- a transform needs to be passed to do this. 
+- I hacked the run file to force it to always write to file, but the verbose output indicated that it was dropping far fewer triples as duplicates than expected (it should have dropped the entire file except for my new storypoint). Further investigation revealed that **the model pulled from staging was pointed at tilesets that were more recent than those versioned in `ris3_points.ttl`**.
+
+This last point meant that updating RIS3 using any code-based approach was impossible, because doing so relied on a versioned `ris3_points.ttl` that was out-of-date compared to whatever unversioned `ris3_points.ttl` had been used to generate the existing model data. Using the older copy would have undone newer changes with no way to recover them.
+
+### Update attempt 3
+
+The storypoint was eventually created using the following incredibly brittle method:
+
+- Add the new storypoint into `ris3_views.py` and run this transform only to generate the triples for the storypoint.
+- Open the triples json and find and copy only the new triples.
+- Open the model pulled down from staging and paste the new triples into it, after making sure that `asOrdinal` wasn't going to collide with any existing values.
+- Also edit the markdown for the sidebar in the model to add a link to the new storypoint.
+- Push the model back up to staging.
+
+(This involved a lot of direct editing of a 2.5MB slab of JSON representing the model, which caused Visual Studio Code to crash twice.)
+
+Obviously I have compounded the original problem here by adding even more unversioned data to the model -- although I have made a PR with an update to `ris3_views`, the generated triples will have the wrong tileset version as there's no good way of setting this on a per-storypoint level yet. Even if there was, though, `ris3_views` is still out-of-sync with `ris3_points.ttl`.
+
+### A note on tile generation
+
+Tile generation from PDFs is usually done On Martin's Computer using a shell script. These shell scripts are undocumented; Martin provided one for RIS3 prior to going on holiday but it started with a huge array of commented-out variables (presumably corresponding to the different tilesets for RIS3) that made things generally rather impenetrable. Now that I have gone through this learning process I *think* I could probably use this to generate some tiles, but when looking at it from the other end it was much easier to let Mark N do it as he knew the AI-native method would work.

```

Comment:
This gets fixed by adopting consistent standards for graphical assembly that doesn't require a human in the middle to work out what the intention is.